src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -1421,6 +1431,16 @@ def __init__(
         else:
             self.path = path
 
+    def _get_file_purpose(self):
+        static_info = self._get_static_info()
+        try:
+            file_purpose = static_info["singletons"]["File"]["commands"][
```

Comment:
This doesn't look right. What's special about singletons? This is separating singletons from named objects for instance, right? Why even do we need to go to that level? Cn we not just access the local static info automatically?

Reply:
@seanpearsonuk 

`_stored_static_info` is empty until we read a case/mesh.

![image](https://github.com/ansys/pyfluent/assets/106588300/b836999f-25c7-44fc-af98-580b29781b05)

`_stored_static_info` is returns value only after reading a case or mesh file. We need file_purpose before execution of command.

We are already using static info using `static_info = self._get_static_info()` 

![image](https://github.com/ansys/pyfluent/assets/106588300/90fb964d-689d-4d35-ba05-6a2d8594aeb4)

According to above static info structure we are accessing `file_purpose`.

Do we have any other way to access file_purpose ?

Reply:
@seanpearsonuk 

As per our latest discussion we are unable to access FileName as follows.

read_mesh = meshing_session.meshing.File.ReadMesh.create_command_instance()`
`read_mesh.FileName` -> gives key error for ReadMesh.

Reply:
Done. Thank you.

Reply:
@hpohekar I frequently get confused when I see resolved conversations that refer to outstanding issues.

Reply:
@seanpearsonuk I'm not sure which changes are you looking at. I have updated the code and latest changes are as follows.

![image](https://github.com/ansys/pyfluent/assets/106588300/59825331-af2a-4fb8-af49-63be6157b06d)

Reply:
@seanpearsonuk https://github.com/ansys/pyfluent/pull/2289/files#diff-c29fea6870a8113071deda7cddca1079fab43d6f0e3229ea44cd9e697f35c6d9

Reply:
Thanks @hpohekar - that looks better. GitHub was still showing me the older version. Apologies.

Reply:
No issues at all.